### DISCUSSION TITLE
SessionUser를 User로 변환해주는 방법

### DISCUSSION THREAD

**Discussion by taesikyoo** [kiworkshop/snowball#132] [Q&A]
![image](https://user-images.githubusercontent.com/37072010/101748265-b01a6480-3b0f-11eb-82a0-7f7d0e3aa32d.png)
위와 같이 NoteController의 createNote 메서드에서 @LoginUser라는 어노테이션을 이용해 세션에서 SessionUser를 받아와 사용하려합니다. Note가 생성될 때 User객체도 Note에 포함되야하기 때문에 SessionUser를 User로 변환하는 작업(이지만 사실은 UserRepository에서 User를 꺼내오는 작업)이 필요할 것 같습니다.

SessionUser는 Serializable을 구현한 dto이고, CustomOAuth2UserService에서 session에 다음과 같이 저장되고 있습니다.
![image](https://user-images.githubusercontent.com/37072010/101748666-3040ca00-3b10-11eb-863c-8b1a9d18ebb4.png)
![image](https://user-images.githubusercontent.com/37072010/101748808-5e260e80-3b10-11eb-87e3-fa090a7a3c5b.png)

어떤 방식으로 SessionUser를 User로 바꿔서 사용하면 좋을까요?
일종의 converter util을 새롭게 만들어서 사용할지, 아니면 user가 필요할때마다 user service에서 받아올지, 아니면 근본적으로 더 나은 방법이 있을지 고민이 됩니다.

코드는 https://github.com/kiworkshop/snowball/tree/feature/8-oauth2-google-login 여기서 확인하실 수 있습니다.

**Comment by kwangilcho** [comment]
<img width="1669" alt="image" src="https://user-images.githubusercontent.com/12382114/101990421-56ba5d00-3cea-11eb-9b6b-9893db128370.png">
이 영역에서 직접 Repository를 접근하거나 하는 일을 할 수는 없을까요?

**Reply by taesikyoo** [reply]
https://github.com/kiworkshop/snowball/commit/6fb171e6247937401bad7181be642a75a8f56309

요렇게 수정해보았습니다! 감사해요😎
